docs/licensing/openrewrite-licensing.md

```diff
@@ -0,0 +1,50 @@
+# OpenRewrite licensing
+
+The core OpenRewrite framework is fully open and Apache licensed, and it will continue to be going forward. If open source software (OSS) authors want to provide software migrations to their users as pure OSS, they can.
+
+OpenRewrite recipes that have significant contributions from Moderne and third-party contributors will be available with the [Moderne Source Available License](https://docs.moderne.io/administrator-documentation/licensing/moderne-source-available-license/). This license precludes others from deriving significant commercial value from OpenRewrite without contributing to the community. Individual organizations can consume and apply these recipes to their source code, but the recipes cannot be put into commercial products or be re-sold.
+
+Moderne and others through partnerships with Moderne can offer commercial recipes.
+
+## Licensing types
+
+The OpenRewrite ecosystem is a marketplace of both free and proprietary recipes. The types of licensing can include:
+
+### Apache License Version 2.0
+
+Core OpenRewrite technology and language libraries in general are fully open sourced under the Apache license. This includes the entirety of [openrewrite/rewrite](https://github.com/openrewrite/rewrite), including rewrite-core and many of the original language bindings (Java, Groovy, Kotlin, XML, Properties, YAML, Terraform, etc.). It also includes the building-block core recipes like `ChangeMethodName`, `ChangeType`, etc. and recipe repositories that are primarily committed to by third parties. This means that framework/library authors can supply OSS migrations to their consumers.
```

Comment:
might be nice to link to https://www.apache.org/licenses/LICENSE-2.0 for those curious to know more about the details